alacritty_terminal/src/term/color.rs

```diff
@@ -11,82 +10,56 @@ use alacritty_config_derive::SerdeReplace;
 
-/// Number of terminal colors.
-pub const COUNT: usize = 269;
-
+// Shim around [`vte::ansi::Rgb`] for custom deserialization and
+// serde replace impl.
 #[derive(SerdeReplace, Debug, Eq, PartialEq, Copy, Clone, Default, Serialize)]
-pub struct Rgb {
-    pub r: u8,
-    pub g: u8,
-    pub b: u8,
-}
+pub struct Rgb(vte::ansi::Rgb);
 
 impl Rgb {
-    /// Implementation of W3C's luminance
-    /// [algorithm](https://www.w3.org/TR/WCAG20/#relativeluminancedef)
-    fn luminance(self) -> f64 {
-        let channel_luminance = |channel| {
-            let channel = channel as f64 / 255.;
-            if channel <= 0.03928 {
-                channel / 12.92
-            } else {
-                f64::powf((channel + 0.055) / 1.055, 2.4)
-            }
-        };
-
-        let r_luminance = channel_luminance(self.r);
-        let g_luminance = channel_luminance(self.g);
-        let b_luminance = channel_luminance(self.b);
+    #[inline]
+    pub const fn new(r: u8, g: u8, b: u8) -> Self {
+        Self(vte::ansi::Rgb { r, g, b })
+    }
 
-        0.2126 * r_luminance + 0.7152 * g_luminance + 0.0722 * b_luminance
+    #[inline]
+    pub fn as_tuple(self) -> (u8, u8, u8) {
+        (self.0.r, self.0.g, self.0.b)
     }
+}
 
-    /// Implementation of [W3C's contrast algorithm].
-    ///
-    /// [W3C's contrast algorithm]: https://www.w3.org/TR/WCAG20/#contrast-ratiodef
-    pub fn contrast(self, other: Rgb) -> f64 {
-        let self_luminance = self.luminance();
-        let other_luminance = other.luminance();
+impl From<vte::ansi::Rgb> for Rgb {
+    fn from(value: vte::ansi::Rgb) -> Self {
+        Self(value)
+    }
+}
 
-        let (darker, lighter) = if self_luminance > other_luminance {
-            (other_luminance, self_luminance)
-        } else {
-            (self_luminance, other_luminance)
-        };
+impl Deref for Rgb {
+    type Target = vte::ansi::Rgb;
 
-        (lighter + 0.05) / (darker + 0.05)
+    fn deref(&self) -> &Self::Target {
+        &self.0
     }
 }
 
-// A multiply function for Rgb, as the default dim is just *2/3.
 impl Mul<f32> for Rgb {
     type Output = Rgb;
 
-    fn mul(self, rhs: f32) -> Rgb {
-        let result = Rgb {
-            r: (f32::from(self.r) * rhs).clamp(0.0, 255.0) as u8,
-            g: (f32::from(self.g) * rhs).clamp(0.0, 255.0) as u8,
-            b: (f32::from(self.b) * rhs).clamp(0.0, 255.0) as u8,
-        };
-
-        trace!("Scaling RGB by {} from {:?} to {:?}", rhs, self, result);
-
-        result
+    fn mul(self, rhs: f32) -> Self::Output {
+        Rgb(self.0 * rhs)
     }
 }
 
 impl Add<Rgb> for Rgb {
     type Output = Rgb;
 
-    fn add(self, rhs: Rgb) -> Rgb {
-        Rgb {
-            r: self.r.saturating_add(rhs.r),
-            g: self.g.saturating_add(rhs.g),
-            b: self.b.saturating_add(rhs.b),
-        }
+    fn add(self, rhs: Rgb) -> Self::Output {
+        Rgb(self.0 + rhs.0)
     }
 }
 
+/// Number of terminal colors.
+pub const COUNT: usize = 269;
```

Comment:
This should stay at the top of the file.

Reply:
Fixed in https://github.com/alacritty/alacritty/pull/6896/commits/d067434b45ddb26b5dd181a98d6fc022962a9a51.